test/gtest/gru_extra.cpp

```diff
@@ -0,0 +1,152 @@
+    switch(prec)
+    {
+    case miopenFloat: params = GRUExtraConfigWithFloat::GetParam(); break;
+    case miopenHalf:
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenHalf, miopenInt8, miopenBFloat16, miopenInt32, miopenDouble "
+                  "data type not supported by "
+                  "rnn_vanilla test";
+
+    default: params = GRUExtraConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<gru_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx900" || devName == "gfx906" || devName == "gfx908" || devName == "gfx90a" ||
+       miopen::StartsWith(devName, "gfx94") || miopen::StartsWith(devName, "gfx103") ||
+       miopen::StartsWith(devName, "gfx110"))
+        return true;
+    else
+        return false;
+}
+
+std::vector<std::string> GetTestCases(void)
+{
+    std::string commonFlags =
+        " --verbose --batch-size 32 --seq-len 3 --batch-seq 32 32 32 --vector-len 128 "
+        "--hidden-size 128 --num-layers 1 --in-mode 0 --bias-mode 0";
+    std::string dir0   = " -dir-mode 0";
+    std::string dir1   = " -dir-mode 1";
+    std::string rnn0   = " --rnn-mode 0";
+    std::string rnn1   = " --rnn-mode 1";
+    std::string no_hx  = " --no-hx";
+    std::string no_hy  = " --no-hy";
+    std::string no_dhx = " --no-dhx";
+    std::string no_dhy = " --no-dhy";
+
+    const std::vector<std::string> test_cases = {
+        // clang-format off
+    			{commonFlags + dir0 + no_hx},
+		{commonFlags + dir0 + no_dhy},
+		{commonFlags + dir0 + no_hx + no_dhy},
+		{commonFlags + dir1 + no_hx},
+		{commonFlags + dir1 + no_dhy},
+		{commonFlags + dir1 + no_hx + no_dhy},
+		{commonFlags + dir0 + no_hy},
+		{commonFlags + dir0 + no_dhx},
+		{commonFlags + dir0 + no_hy + no_dhx},
+		{commonFlags + dir1 + no_hy},
+		{commonFlags + dir1 + no_dhx},
+		{commonFlags + dir1 + no_hy + no_dhx},
+		{commonFlags + dir0 + no_hx + no_dhy + no_hy + no_dhx},
+		{commonFlags + dir1 + no_hx + no_dhy + no_hy + no_dhx}
+        // clang-format on
+    };
+
+    return test_cases;
+}
+
+} // namespace gru_extra
+using namespace gru_extra;
+
+TEST_P(GRUExtraConfigWithFloat, FloatTest_gru_extra)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenFloat);
```

Comment:
When MIOPEN_TEST_ALL is set (CTest mode), the test should be run only if (MIOPEN_TEST_FLOAT_ARG == "--float"). Currently this is not so. Valid code:
```C++
if(IsTestSupportedForDevice(handle) //
   && (IsUnset(ENV(MIOPEN_TEST_ALL)) //
       || (IsEnabled(ENV(MIOPEN_TEST_ALL)) //
           && GetValue(ENV(MIOPEN_TEST_FLOAT_ARG)) == "--float")))
    Run2dDriver(miopenFloat);
else
    GTEST_SKIP();
```

Reply:
[resolved]